assets/bower_components/three-js/three-js.html

```diff
@@ -1241,6 +1241,152 @@
+            THREE[this.format], THREE[this.type]);
+        if (this.magfilter) {
+          this.object.magFilter = THREE[this.magfilter];
+        }
+        if (this.minfilter) {
+          this.object.minFilter = THREE[this.minfilter];
+        }
+        this.object.unpackAlignment = 1;
+        this.object.needsUpdate = true;
+        this.addToParent();
+      },
+      addToParent: function () {
+        Polymer.dom(this).parentNode.uniforms[this.name] = {type: "t", value: this.object};
+      },
+      attached: function () {
+        this.async(function (){
+          this.newDataTexture();
+        });
+      },
+      imageChanged: function() {
+        if (this.object) {
+          this.object.image.data = this.b64decode(this.data);
+          this.object.image.width = this.width;
+          this.object.image.height = this.height;
+          this.object.needsUpdate = true;
+        }
+      },
+ b64decode: function (base64) {
+  // lightly adapted from Niklas
+
+  /*
+   * base64-arraybuffer
+   * https://github.com/niklasvh/base64-arraybuffer
+   *
+   * Copyright (c) 2012 Niklas von Hertzen
+   * Licensed under the MIT license.
+   */
+  var chars =
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+  var bufferLength = base64.length * 0.75,
+    len = base64.length,
+    i, p = 0,
+    encoded1, encoded2, encoded3, encoded4;
+
+  if (base64[base64.length - 1] === "=") {
+    bufferLength--;
+    if (base64[base64.length - 2] === "=") {
+      bufferLength--;
+    }
+  }
+
+  var bytes = new Uint8Array(bufferLength);
+
+  for (i = 0; i < len; i += 4) {
+    encoded1 = chars.indexOf(base64[i]);
+    encoded2 = chars.indexOf(base64[i + 1]);
+    encoded3 = chars.indexOf(base64[i + 2]);
+    encoded4 = chars.indexOf(base64[i + 3]);
+
+    bytes[p++] = (encoded1 << 2) | (encoded2 >> 4);
+    bytes[p++] = ((encoded2 & 15) << 4) | (encoded3 >> 2);
+    bytes[p++] = ((encoded3 & 3) << 6) | (encoded4 & 63);
+  }
+
+  return bytes;
+},
+    });
+  </script>
+</dom-module>
+
+<dom-module id="three-js-shader-material">
+  <script>
+    Polymer({
+      is: 'three-js-shader-material',
+      properties: {
+        vert: {
+          type: String,
+        },
+        frag: {
+          type: String,
+        },
+        defines: {
+          type: Object,
+        },
+        uniforms: {
+          type: Object,
+        },
+      },
+      newMaterial: function () {
+        this.object = new THREE.ShaderMaterial({
+          vertexShader: this.vert,
+          fragmentShader: this.frag,
+          defines: this.defines,
+          uniforms: this.uniforms,
+        });
+        Polymer.dom(this).parentNode.addMaterial(this.object);
+      },
+      attached: function () {
+        this.async(function (){
+            this.newMaterial();
+        });
+      }
+    });
+
+  PropertyHook.setupHook("three-js-shader-material", "uniforms",
```

Comment:
I think @rohitvarkey knows that Polymer cannot listen to changes to `elem.prop.subProp`, the `hooks` thing lets you do that.